ggml-cuda.cu

```diff
@@ -24,23 +51,23 @@ static_assert(sizeof(block_q4_1) == sizeof(float) * 2 + QK4_1 / 2, "wrong q4_1 b
 
 #define QK4_2 16
 typedef struct {
-    __half  d;              // delta
+    half  d;                // delta
     uint8_t qs[QK4_2 / 2];  // nibbles / quants
 } block_q4_2;
 static_assert(sizeof(block_q4_2) == sizeof(ggml_fp16_t) + QK4_2 / 2, "wrong q4_2 block size/padding");
 
 #define QK5_0 32
 typedef struct {
-    __half d;               // delta
+    half d;                 // delta
     uint8_t qh[4];          // 5-th bit of quants
     uint8_t qs[QK5_0 / 2];  // nibbles / quants
 } block_q5_0;
 static_assert(sizeof(block_q5_0) == sizeof(ggml_fp16_t) + sizeof(uint32_t) + QK5_0 / 2, "wrong q5_0 block size/padding");
 
 #define QK5_1 32
 typedef struct {
-    __half d;               // delta
-    __half m;               // min
+    half d;                 // delta
+    half m;                 // min
     uint32_t qh;            // 5-th bit of quants
```

Comment:
At some point, should sync the CUDA `block_q5_1` with the CPU one:

https://github.com/ggerganov/llama.cpp/blob/c0335b51f959ddf8c7b58bf497d10e5dc4730267/ggml.c#L736

Reply:
I am not entirely sure why this isn't the case already, did you have any problems with alignment or anything else?

Reply:
I updated it in the same way as q5_0 and didn't notice any issues.

Reply:
For `Q5_1` it works both ways.
For `Q5_0`, the `uint32_t` way does not work due to alignment issues, so we changed `Q5_1` to `uint8_t[4]` for consistency